### PR TITLE
GitHub action to update jdks.yaml

### PR DESCRIPTION
**Pull request by blindpirate**
Fixes https://github.com/gradle/gradle-private/issues/4506

See here for the example usage: https://github.com/blindpirate/gradle/blob/master/.github/workflows/update-jdks.yml

This is the example PR created: https://github.com/blindpirate/gradle/pull/30

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,7 +1,152 @@
...
+  }
+  return parseInt(match[1], 10)
+}
+
+async function getLatestTemurinVersion(jdk: Jdk): Promise<{ version: string; sha256: string }> {
+  const majorVersion = extractMajorVersion(jdk.version)
+  const targetArch = getOrError(ARCH_MAPPING, jdk.arch)
+  const targetOs = getOrError(OS_MAPPING, jdk.os)
+  const apiUrl = `https://api.adoptium.net/v3/assets/latest/${majorVersion}/hotspot?os=${targetOs}&architecture=${targetArch}`
+  const response = await axios.get(apiUrl)
```
Comment (ov7a) [inline src/index.ts:141]: 🤔 do we need an extra lib for that? `fetch` can do the work

File: src/index.ts [hunk 2]
```diff
@@ -1,7 +1,152 @@
...
+
+  const latestAsset = response.data[0]
+
+  if (latestAsset) {
+    return { version: latestAsset.release_name, sha256: latestAsset.binary.package.checksum }
```
Comment (ov7a) [inline src/index.ts:146]: ❌ I think we should check that those values are present to avoid "undefined"

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,167 @@
-/**
- * The entrypoint for the action.
+import * as core from '@actions/core'
```
Comment (cobexer) [inline src/index.ts:1]: ❌ Should have our usual copyright header

File: src/index.ts [hunk 4]
```diff
@@ -1,7 +1,167 @@
...
+  const data = await response.json()
+  const latestAsset = data[0]
+
+  if (
+    !latestAsset ||
+    !latestAsset.release_name ||
+    !latestAsset.binary?.package?.checksum
+  ) {
+    throw new Error(
+      `Invalid data structure: expected release_name and binary.package.checksum in the response for JDK ${majorVersion}`
```
Comment (cobexer) [inline src/index.ts:155]: 💅 we should log the response here to simplify debugging

**Review by cobexer — CHANGES_REQUESTED** [review]

**Review by cobexer — APPROVED** [review]
LGTM

**Pull request merged**